packages/@uppy/core/src/Uppy.ts

```diff
@@ -32,27 +36,29 @@ import type UIPlugin from './UIPlugin.ts'
 import type { Restrictions } from './Restricter.ts'
 import type { PluginOpts } from './BasePlugin.ts'
 
-type Processor = (fileIDs: string[], uploadID: string) => Promise<void>
+type Processor = (fileIDs: string[], uploadID: string) => Promise<void> | void
 
 type FileRemoveReason = 'removed-by-user' | 'cancel-all'
 
 type LogLevel = 'info' | 'warning' | 'error' | 'success'
 
-type UploadHandler = (fileIDs: string[]) => Promise<void>
-
 type UnknownPlugin<M extends Meta, B extends Body> = InstanceType<
   typeof BasePlugin<any, M, B> | typeof UIPlugin<any, M, B>
 >
 
+// The user facing type for UppyFile used in uppy.addFile() and uppy.setOptions()
 type MinimalRequiredUppyFile<M extends Meta, B extends Body> = Required<
   Pick<UppyFile<M, B>, 'name' | 'data' | 'type' | 'source'>
 > &
-  Partial<Omit<UppyFile<M, B>, 'name' | 'data' | 'type' | 'source'>>
+  Partial<
+    Omit<UppyFile<M, B>, 'name' | 'data' | 'type' | 'source' | 'meta'>
+  > & { meta?: M }
```

Comment:
Why is `meta` omitted on line 54 and re-added on line 55? Can you add a comment?